### DISCUSSION TITLE
How I can customize query on update model?

### DISCUSSION THREAD

**Discussion by adonskoi** [aminalaee/sqladmin#676] [Q&A]
On update model `await Query(self).update(pk, data)` has call, next `_update_async` with lines:
```
for relation in self.model_view._relations:
    stmt = stmt.options(joinedload(relation))
```
I get here tremendously complex query with many joins. Are there some workarounds from this?

This does not seem to affect the model update `column_details_exclude_list, form_excluded_columns`

**Comment by aminalaee** [comment]
Hmm...in the async SQLAlchemy there's no way to update without the JOIN-loaded as far as I know.

**Reply by adonskoi** [reply]
You absolutely right in general case. If you want to update many-to-many relation without join - you get error. But in case if I want to update only table field, for example username:str in table users. I can do it without joins.
Comment this lines in `_queries.py::Query._update_async` solve my problem - I get query without any joins, but this is not a solution:
```
for relation in self.model_view._relations:
     stmt = stmt.options(joinedload(relation))
```

**Reply by aminalaee** [reply]
Yeah thats true. I think the solution can be to check if relationship attributes are changed and only in that case join the tables.
